onnx-frontend.py

```diff
@@ -0,0 +1,524 @@
+
+    mo.md("""
+    We then use MLIR to bufferize our function.
+    """)
+    return bufferized_module,
+
+
+@app.cell
+def __(MLIROptPass, bufferized_module, ctx):
+    scf_module = bufferized_module.clone()
+
+    MLIROptPass(
+        arguments=["--convert-linalg-to-loops", "--lower-affine", "--canonicalize"]
+    ).apply(ctx, scf_module)
+
+    print(scf_module)
+    return scf_module,
+
+
+@app.cell
+def __(ctx, scf_module):
+    from xdsl.backend.riscv.lowering import (
+        convert_arith_to_riscv,
+        convert_func_to_riscv_func,
+        convert_memref_to_riscv,
+        convert_scf_to_riscv_scf,
+    )
+    from xdsl.passes import PipelinePass
+    from xdsl.transforms import reconcile_unrealized_casts
+
+    riscv_module = scf_module.clone()
+
+    lower_to_riscv = PipelinePass(
+        [
+            convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass(),
+            convert_memref_to_riscv.ConvertMemrefToRiscvPass(),
+            convert_arith_to_riscv.ConvertArithToRiscvPass(),
+            convert_scf_to_riscv_scf.ConvertScfToRiscvPass(),
+            reconcile_unrealized_casts.ReconcileUnrealizedCastsPass(),
+        ]
+    ).apply(ctx, riscv_module)
+
+    print(riscv_module)
+    return (
+        PipelinePass,
+        convert_arith_to_riscv,
+        convert_func_to_riscv_func,
+        convert_memref_to_riscv,
+        convert_scf_to_riscv_scf,
+        lower_to_riscv,
+        reconcile_unrealized_casts,
+        riscv_module,
+    )
+
+
+@app.cell
+def __(PipelinePass, ctx, riscv_module):
+    from xdsl.backend.riscv.lowering.convert_snitch_stream_to_snitch import (
+        ConvertSnitchStreamToSnitch,
+    )
+    from xdsl.transforms.canonicalize import CanonicalizePass
+    from xdsl.transforms.lower_snitch import LowerSnitchPass
+    from xdsl.transforms.riscv_register_allocation import RISCVRegisterAllocation
+    from xdsl.transforms.riscv_scf_loop_range_folding import (
+        RiscvScfLoopRangeFoldingPass,
+    )
+    from xdsl.transforms.snitch_register_allocation import SnitchRegisterAllocation
+
+    regalloc_module = riscv_module.clone()
+
+    PipelinePass(
+        [
+            RISCVRegisterAllocation(),
+            CanonicalizePass(),
+        ]
+    ).apply(ctx, regalloc_module)
+
+    print(regalloc_module)
+    return (
+        CanonicalizePass,
+        ConvertSnitchStreamToSnitch,
+        LowerSnitchPass,
+        RISCVRegisterAllocation,
+        RiscvScfLoopRangeFoldingPass,
+        SnitchRegisterAllocation,
+        regalloc_module,
+    )
+
+
+@app.cell
+def __(CanonicalizePass, ctx, regalloc_module):
+    from xdsl.backend.riscv.lowering.convert_riscv_scf_to_riscv_cf import (
+        ConvertRiscvScfToRiscvCfPass,
+    )
+    from xdsl.dialects.riscv import riscv_code
+
+    assembly_module = regalloc_module.clone()
+
+    ConvertRiscvScfToRiscvCfPass().apply(ctx, assembly_module)
+    CanonicalizePass().apply(ctx, assembly_module)
+
+    str(assembly_module)
```

Comment:
```suggestion
    print(assembly_module)
```

This would be more consistent with the remaining IR.